lib/_http_outgoing.js

```diff
@@ -673,6 +675,40 @@ OutgoingMessage.prototype.setHeader = function setHeader(name, value) {
   return this;
 };
 
+OutgoingMessage.prototype.setHeaders = function setHeaders(headers) {
+  if (this._header) {
+    throw new ERR_HTTP_HEADERS_SENT('set');
+  }
+  let k;
+
+  if (headers instanceof Headers) {
+    const keys = [...headers.keys()];
```

Comment:
I wonder if here,, since we have to copy to an array (which probably iterates it) and then doing a C++ for-loop, it is just easier and faster to do a `for-of`.